app/models/product.rb

```diff
@@ -1,12 +1,34 @@
+class UrlValidator < ActiveModel::EachValidator
+  def validate_each(record, attr, value)
+    unless value =~ %r{\.(gif|jpg|png)\z}i
+      record.errors.add attr, 'must be a URL for GIF, JPG or PNG images'
+    end
+  end
+end
+
+class PriceValidator < ActiveModel::Validator
+  def validate(record)
+    if record[:price] < record[:discount_price]
+      record.errors.add :price, "should be greater than #{record[:discount_price]}"
+    end
+  end
+end
+
 class Product < ApplicationRecord
 
+  validates :permalink, uniqueness:true
+  validates :permalink, format: {
+    with: %r{\A[A-Za-z0-9-]+\Z},
+    message: 'should not contains any special characters'
+  }
+  validates :words_in_permalink, length: {minimum: 3}
+  validates :words_in_description , length: {in: 5..10}
```

Comment:
we should run validations on model attributes. Lets add custom methods for validating these.

Reply:
done